packages/experiments/src/components/Recommendation/Recommendation.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+
+/* Dependent Components */
+import { PrimaryButton } from 'office-ui-fabric-react/lib/Button';
+import { CardFrame, ICardDropDownOption } from '../Card/CardFrame';
+
+/* Types for props and styles */
+import { IRecommendationProps, IRecommendationStyles } from './Recommendation.types';
+
+/* Styles for CardComponent and Recommendation Card */
+import { CardComponentStyles, getStyles } from './Recommendation.styles';
+
+/* Utilities */
+import { classNamesFunction } from 'office-ui-fabric-react/lib/Utilities';
+
+export class Recommendation extends React.Component<IRecommendationProps, {}> {
+  /* Default Props */
+  public static defaultProps = {
+    recommendationButtonLocalizedName: 'View Recommendation',
+    recommendationButtonAriaDescription: 'Click to View Recommendation',
+    dismissRecommendationLocalizedName: 'Dismiss',
+    dismissRecommendationAriaLabel: 'Dismiss Recommendation'
+  };
+
+  private recommendationMenuItems: ICardDropDownOption[];
+
+  constructor(props: IRecommendationProps) {
+    super(props);
+  }
+
+  public render(): JSX.Element {
+    const {
+      recommendationBarTitle,
+      recommendationDescriptionHeader,
+      recommendationDescription,
+      recommendationButtonLocalizedName,
+      recommendationButtonAriaDescription,
+      handleViewRecommendationClick,
+      dismissRecommendationLocalizedName,
+      dismissRecommendationAriaLabel,
+      handleDismissRecommendationClick
+    } = this.props;
+
+    const getClassNames = classNamesFunction<{}, IRecommendationStyles>();
+    const classNames = getClassNames(getStyles!);
+    this.recommendationMenuItems = [
```

Comment:
>recommendationMenuItems [](start = 9, length = 23)

why is it a private variable ? why not local const ?

Reply:
does it make any difference ? Private variable because i want it to be constrained with in the class and shouldn't be accessed outside of it.

Reply:
does it make any difference ? Private variable because i want it to be constrained with in the class and shouldn't be accessed outside of it.

---
In reply to: [196372995](https://github.com/OfficeDev/office-ui-fabric-react/pull/5252#discussion_r196372995) [](ancestors = 196372995)

Reply:
1. the scope of the variable is only in render method, not used outside. possibility of using it outside will result in undefined
2. it is assigned and not modified, so it is better to be a const

---
In reply to: [196537890](https://github.com/OfficeDev/office-ui-fabric-react/pull/5252#discussion_r196537890) [](ancestors = 196537890,196372995)